airflow/contrib/hooks/neo4j_hook.py

```diff
@@ -0,0 +1,106 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""This hook provides minimal thin wrapper around the neo4j python library to provide query execution"""
+from typing import Optional
+from neo4j import BoltStatementResult, Driver, GraphDatabase, Session
+from airflow.exceptions import AirflowException
+from airflow.hooks.base_hook import BaseHook
+
+
+class Neo4JHook(BaseHook):
+    """This class enables the neo4j operator to execute queries against a configured neo4j server.
+    It requires the configuration name as set in Airflow -> Connections ->
+    :param n4j_conn_id:
+    :type str:
+    """
+    _n4j_conn_id = None
+
+    template_fields = ['n4j_conn_id']
+
+    def __init__(self, n4j_conn_id: str = 'n4j_default', *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._n4j_conn_id = n4j_conn_id
+
+    @staticmethod
+    def get_config(n4j_conn_id: Optional[str]) -> dict:
+        """
+        Obtain the Username + Password from the Airflow connection definition
+        Store them in _config dictionary as:
+        *credentials* -- a tuple of username/password eg. ("username", "password")
+        *host* -- String for Neo4J URI eg. "bolt://1.1.1.1:7687"
+        :param n4j_conn_id: Name of connection configured in Airflow
+        :type n4j_conn_id: str
+        :return: dictionary with configuration values
+        :rtype dict
+        """
+        # Initialize with empty dictionary
+        config: dict = {}
+        if n4j_conn_id is not None:
+            connection_object = Neo4JHook.get_connection(n4j_conn_id)
+            if connection_object.login and connection_object.host:
+                config['credentials'] = connection_object.login, connection_object.password
+                config['host'] = "bolt://{0}:{1}".format(connection_object.host, connection_object.port)
+        else:
+            raise AirflowException("No Neo4J connection: {}".format(n4j_conn_id))
```

Comment:
I would do this like:
```python
if n4j_conn_id is None:
    raise AirflowException("No Neo4J connection: {}".format(n4j_conn_id))

connection_object = Neo4JHook.get_connection(n4j_conn_id)
if connection_object.login and connection_object.host:
    ...
```
It's simpler to read imho. Additionally, will the exception always be formatted with `None`?

Reply:
I have resolved this by making it require a string, `None` is not accepted by the function. As a result, no exception is to be thrown.